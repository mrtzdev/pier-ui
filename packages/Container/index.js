import Container from "./src/Container";

export default Container;
