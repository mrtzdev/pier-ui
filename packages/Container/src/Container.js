import PropTypes from "prop-types";

import defaultTheme from "../../Theme/theme";
import { useTheme } from "../../UiProvider";

const propTypes = {
  className: PropTypes.string,
  as: PropTypes.string,
  children: PropTypes.any,
  maxWidth: PropTypes.string,
};

const defaultProps = {
  as: "div",
};

const Container = (props) => {
  const { className, maxWidth, as, children, ...restProps } = props;
  const Component = as;

  const customTheme = useTheme();
  const theme = customTheme ? customTheme : defaultTheme;

  return (
    <>
      <Component className={`container ${className ? className : ""}`}>
        {children}
      </Component>
      <style jsx>{`
        .container {
          width: 100%;
          margin: 0 auto;
          position: relative;
          ${maxWidth
            ? "max-width:" + maxWidth
            : "max-width:" + theme.maxWidth.lg};
        }
      `}</style>
    </>
  );
};

export default Container;

Container.propTypes = propTypes;
Container.defaultProps = defaultProps;
