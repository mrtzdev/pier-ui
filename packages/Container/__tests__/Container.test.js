/**
 * @jest-environment jsdom
 */

import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import Container from "../index";

describe("Container", () => {
  it("should render Container correctly", () => {
    render(<Container></Container>);
    // screen.debug();
  });

  it("Renders with a custom className ", () => {
    const { container } = render(
      <Container className="custom-class"> </Container>
    );
    expect(container.firstChild).toHaveClass("custom-class");
  });

  it("should render empty Container correctly", () => {
    expect(<Container />).toMatchSnapshot();
  });
});
