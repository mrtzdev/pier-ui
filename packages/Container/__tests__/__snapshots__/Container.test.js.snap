// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render empty Container correctly 1`] = `
<Container
  as="div"
/>
`;
